storage/src/checker/mod.rs

```diff
@@ -261,6 +259,16 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
         let (area_index, area_size) = self.area_index_and_size(subtrie_root_address)?;
         let (node, node_bytes) = self.read_node_with_num_bytes_from_disk(subtrie_root_address)?;
 
+        // check if the node fits in the area, equal is not allowed due to 1-byte area size index
```

Comment:
Why isn't `read_node_with_num_bytes_from_disk` including the extra byte? Are there any callers that aren't going to add 1?

Seems like it might be better to pass in the maximum allowed size to `read_node_with_num_bytes_from_disk` so it can stop once it hits the end of the area and return an error, rather than returning the number of bytes and checking. Are there any callers that don't know how big the area is supposed to be?